Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java

```diff
@@ -707,41 +706,41 @@ public Process convertDtoToBean(ProcessDTO processDTO) throws DAOException {
     }
 
     @Override
-    public ProcessDTO convertJSONObjectToDTO(JSONObject jsonObject, boolean related) throws DataException {
+    public ProcessDTO convertJSONObjectToDTO(JsonObject jsonObject, boolean related) throws DataException {
         ProcessDTO processDTO = new ProcessDTO();
         processDTO.setId(getIdFromJSONObject(jsonObject));
-        JSONObject processJSONObject = getSource(jsonObject);
-        processDTO.setTitle(getStringPropertyForDTO(processJSONObject, "title"));
-        processDTO.setOutputName(getStringPropertyForDTO(processJSONObject, "outputName"));
-        processDTO.setWikiField(getStringPropertyForDTO(processJSONObject, "wikiField"));
-        processDTO.setCreationDate(getStringPropertyForDTO(processJSONObject, "creationDate"));
+        JsonObject processJSONObject = jsonObject.getJsonObject("_source");
+        processDTO.setTitle(processJSONObject.getString("title"));
+        processDTO.setOutputName(processJSONObject.getString("outputName"));
+        processDTO.setWikiField(processJSONObject.getString("wikiField"));
+        processDTO.setCreationDate(processJSONObject.getString("creationDate"));
         processDTO.setPropertiesSize(getSizeOfRelatedPropertyForDTO(processJSONObject, "properties"));
         processDTO.setProperties(
                 convertRelatedJSONObjectToDTO(processJSONObject, "properties", serviceManager.getPropertyService()));
         processDTO.setSortedCorrectionSolutionMessages(getSortedCorrectionSolutionMessages(processDTO));
-        processDTO.setSortHelperArticles(getIntegerPropertyForDTO(processJSONObject, "sortHelperArticles"));
-        processDTO.setSortHelperDocstructs(getIntegerPropertyForDTO(processJSONObject, "sortHelperDocstructs"));
-        processDTO.setSortHelperImages(getIntegerPropertyForDTO(processJSONObject, "sortHelperImages"));
-        processDTO.setSortHelperMetadata(getIntegerPropertyForDTO(processJSONObject, "sortHelperMetadata"));
-        //TODO: remove access to database here, store path in index
-        processDTO.setTifDirectoryExists(false);
-        //processDTO.setTifDirectoryExists(
-        //checkIfTifDirectoryExists(processDTO.getId(), processDTO.getTitle(), processDTO.getProcessBaseUri()));
+        processDTO.setSortHelperArticles(processJSONObject.getInt("sortHelperArticles"));
+        processDTO.setSortHelperDocstructs(processJSONObject.getInt("sortHelperDocstructs"));
+        processDTO.setSortHelperImages(processJSONObject.getInt("sortHelperImages"));
+        processDTO.setSortHelperMetadata(processJSONObject.getInt("sortHelperMetadata"));
+        processDTO.setTifDirectoryExists(
```

Comment:
In a previous commit you commeted this call out as it takes a long time and slowed down the displaying of list entries. Is this now solved?

Reply:
Yes, this one accesses database only if value is empty.